File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    int today = 1;

    for (int i = 0; i < progresses.size(); i++) {
        int need = ceil((100 - progresses[i]) / (float)speeds[i]);

        if (answer.empty() || need > today)
            answer.push_back(1);
        else
            answer.back()++;
        
         if (today < need)
            today = need;
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public Stack<Integer> solution(int[] progresses, int[] speeds) {  
        int len = speeds.length;
        int[] arr = new int[len];
        
        for (int i = 0; i < len; i++) {
            arr[i] = (int) Math.ceil((100 - progresses[i]) / (double)speeds[i]);
        }

        Stack<Integer> stack = new Stack<>();
        int tmp = -1;
        
        for (int i = 0; i < len; i++) {
            if (tmp < arr[i]) {
                tmp = arr[i];
                stack.push(1);
            }
            else {
                int top = stack.peek();
                stack.pop();
                stack.push(top+1);
            }
        }
        
        return stack;
    }
}